src/MyArrayList.java

```diff
@@ -0,0 +1,325 @@
+    }
+
+    @Override
+    public Object set(int index, Object elem) {
+
+        Object temp;
+
+        if (index >= array.length) {
+            expand(index);
+            array[index] = elem;
+            temp = null;            //здесь мог быть Exception
+
+        } else {
+            temp = array[index];
+            array[index] = elem;
+        }
+
+        count = index > count ? ++index : count;
+
+        return temp;
+    }
+
+    @Override
+    public boolean remove(Object elem) {
+        int temp = indexOf(elem);
+
+        if (temp == -1) {
+            return false;
+        }
+
+        else {
+            remove(temp);
+            return true;
+        }
+
+    }
+
+    @Override
+    public Object remove(int index) {
+        Object elem;
+        try {
+            elem = this.array[index];        //здесь мог быть Exception
+            shiftLeft(index);
+            count--;
+
+
+        } catch (ArrayIndexOutOfBoundsException e) {
+            e.printStackTrace();
+            throw new IndexOutOfBoundsException(index);
+        }
+
+        return elem;
+    }
+
+    @Override
+    public void clear() {
+        Object[] array = new Object[this.array.length];
+        Arrays.fill(array, 0, this.array.length, null);
+        this.array = array;
+        count = 0;
+    }
+
+    @Override
+    public Object get(int index) {
+        Object elem;
+        try {
+            elem = array[index];
+        } catch (IndexOutOfBoundsException e) {
+            e.printStackTrace();
+            throw new IndexOutOfBoundsException(index);
+        }
+        return elem;
+    }
+
+    @Override
+    public int indexOf(Object elem) {
+        for (int i = 0; i < array.length; i++) {
+            if (elem == null) {
+                if (array[i] == null) {
+                    return i;
+                }
+            }
+
+         else {
+              if (elem.equals(array[i])) {
+                  return i;
+                }
+            }
+
+        }
+
+        return -1;
+    }
+
+    @Override
+    public boolean contains(Object elem) {
+        return indexOf(elem) != -1;
+    }
+
+    @Override
+    public int size() {
+        return count;
+    }
+
+    @Override
+    public Iterator iterator() {
+        return new MyIterator();
+    }
+
+
+    private class MyIterator implements Iterator {
+
+        private int nextCount = 0;
+        private int previousCount = -1;
+
+        @Override
+        public boolean hasNext() {
+            return count > nextCount;
+        }
+
+        @Override
+        public Object next() {
+            Object elem = array[nextCount++];          //здесь может быть Exception
+            previousCount = nextCount - 1;
+            return elem;
+        }
+
+        @Override
+        public void remove() {
+            MyArrayList.this.remove(previousCount);           //здесь может быть ArrayIndexOutOfBoundsException
+            previousCount = -1;
+        }
```

Comment:
Both remove() and next() are implemented not fully correct. There's a condition: within the current iteration remove() could be called after next() was called. For now there are no any of these checks. Please do